api/src/main/java/org/pmiops/workbench/cohortbuilder/SubjectCounter.java

```diff
@@ -0,0 +1,88 @@
+package org.pmiops.workbench.cohortbuilder;
+
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.QueryRequest;
+import org.pmiops.workbench.cohortbuilder.querybuilder.FactoryKey;
+import org.pmiops.workbench.cohortbuilder.querybuilder.QueryParameters;
+import org.pmiops.workbench.model.SearchGroup;
+import org.pmiops.workbench.model.SearchGroupItem;
+import org.pmiops.workbench.model.SearchRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides counts of unique subjects
+ * defined by the provided {@link SearchRequest}.
+ */
+@Service
+public class SubjectCounter {
+
+    private static final String COUNT_SQL_TEMPLATE =
+            "select count(distinct person_id) as count\n" +
+                    "  from `${projectId}.${dataSetId}.person` person\n" +
+                    " where 1 = 1\n";
```

Comment:
Nit: get rid of 1=1 here and the " and " from INCLUDE_SQL_TEMPLATE and have below:

StringJoiner joiner = new StringJoiner(" and ")
for (SearchGroup ...)

  joiner.add(INCLUDE_SQL_TEMPLATE.replace("${includeSql}", String.join(UNION_TEMPLATE, queryParts)))

finalSql = COUNT_SQL_TEMPLATE + joiner.toString()

(or something like that). Having 1=1 in the resulting SQL is kinda kooky. :)

Reply:
Wait... you don't like kooky? Hehe... agreed it is kinda kooky and i will change that.